LIBVmotor0.11/LIBVmotor0.11.ino

```diff
@@ -216,34 +292,157 @@ void Alarm(bool _stat, int type){
 
 //Calculate the Breath timing from Setting
 void BPM_Timing(){
-  timeIn = TI;
-  timeOut = (TI*IE*0.5)/2.0;
-  timeRest = timeOut; 
-  TV2RPB = (TV/TV_MAX)*RPB;
-  speedIn = TV2RPB*1000.0/(float)timeIn;
-  speedOut = -(TV2RPB*1000.0/(float)timeOut);
+  if (modeVCV)
+  {
+    /*New variable to OLD system for demo*/
+    TV = VCVsetting.TV;
+    IE = VCVsetting.IE;
+    TI = round(60000.0 / (VCVsetting.RR*(1+IE)));
+    /*New variable to OLD system for demo*/
+
+    timeIn = TI;
+    timeOut = (TI*IE)/2.0;
+    timeRest = timeOut; 
+    TV2RPB = min(camPosition(TV), RPB);
+    speedIn = TV2RPB*1000.0/(float)timeIn;
+    speedOut = -(TV2RPB*1000.0/(float)timeOut);
+    Serial.print("TV%: ");
+    Serial.println(TV/TV_MAX);
+  }
+
+}
+
+float camPosition(int _tv){
+	switch (_tv) {
+      case 100:
+        return 0.165;//
+        break;
+        
+      case 150:
+        return 0.18;//
+        break;
+        
+      case 200:
+        return 0.195;//
+        break;
+        
+	    case 250:
+	      return 0.205;//
+	      break;
+
+	    case 300:
+	      return 0.223;//
+	    break;
+
+	    case 350:
+	      return 0.231;//
+	      break;
+
+	    case 400:
+	      return 0.245;//
+	      break;
+
+	    case 450:
+	      return 0.253;//
+	      break;
+
+	    case 500:
+	      return 0.27;//
+	      break;
+
+	    case 550:
+	      return 0.28;//
+	      break;
+
+	    case 600:
+	      return 0.29;//
+	      break;
+
+	    case 650:
+	      return 0.30;//
+	      break;
+
+	    case 700:
+	      return 0.31;//
+	      break;
+
+	    case 750:
+	      return 0.32;//
+	      break;
+
+	    case 800:
+	      return 0.335;//0.335
+	      break;
+
+	    default:
+	      return 0.23;
+	}
+
 }
 
 //I2C Receive Data and update to EEPROM
 void receiveEvent(int howMany) {
+  uint8_t _func = Wire.read();
+  switch (_func) {
+      case I2C_CMD_MOTOR:    //Read as motor command
+        breathing = (bool)Wire.read();
+        break;
 
+      case I2C_DATA_SETTING: // Read as setting data
+        modeVCV = (bool)Wire.read();
+        VCVsetting.IP = Wire.read();
+        VCVsetting.TV = Wire.read()/singleByteConst.TV;
+        VCVsetting.RR = Wire.read();
+        VCVsetting.IE = Wire.read()/singleByteConst.IE;
+        VCVsetting.PEEP = Wire.read();
+        BPAPsetting.IP = Wire.read();
+        BPAPsetting.PEEP = Wire.read();
+        BPAPsetting.TP = Wire.read();
+        BPAPsetting.FST = Wire.read()/singleByteConst.FST;
+        break;
+  }
   //Read each byte from master
-  TV = Wire.read()*10;
-  TI = Wire.read()*100;
-  IE = Wire.read();
+  
+
```

Comment:
Since Serial prints are slow operations that might affect your loop speed (and therefore PID tuning), these should be inside an "if(DEBUG)", and then define DEGBUG up a the top of the sketch.

And then make sure any PID tuning that you're doing is done with DEBUG mode off.